app/models/tag.rb

```diff
@@ -364,14 +364,13 @@ def self.tagged_node_count(tag_name)
 
   def self.related(tag_name, count = 5)
     Rails.cache.fetch("related-tags/#{tag_name}/#{count}", expires_in: 1.weeks) do
-      nids = NodeTag.joins(:tag)
-                     .where(Tag.table_name => { name: tag_name })
-                     .select(:nid)
-
-      # sort them by how often they co-occur:
-      nids = nids.group_by{ |v| v }.map{ |k, v| [k, v.size] }
-      nids = nids.collect(&:first)[0..4]
-                 .collect(&:nid) # take top 5
+      nids = NodeTag.joins(:tag, :node)
+                    .where(Node.table_name => { status: 1 })
+                    .where(Tag.table_name => { name: tag_name })
+                    .group(:nid)
+                    .order(NodeTag.arel_table[:nid].count.desc)
```

Comment:
Notes on this `arel_table` usage: https://thoughtbot.com/blog/using-arel-to-compose-sql-queries https://gist.github.com/abelards/b1aa348c802db62e011e8aa472ad4247